pkg/runtime/cli.go

```diff
@@ -55,6 +55,7 @@ func FromFlags() (*DaprRuntime, error) {
 	profilePort := flag.String("profile-port", strconv.Itoa(DefaultProfilePort), "The port for the profile server")
 	appProtocol := flag.String("app-protocol", string(HTTPProtocol), "Protocol for the application: grpc or http")
 	componentsPath := flag.String("components-path", "", "Path for components directory. If empty, components will not be loaded. Self-hosted mode only")
+	dynamicComponentsPath := flag.String("dynamic-components-path", "", "Path of components directory from where components are loaded dynamically. If empty, dynamic loading of components will be disabled. Self-hosted mode")
```

Comment:
Do we have a reason to not reuse the `components-path` flag? Is there a usecase when dynamic is not desired?

Reply:
@mcandeia - The intention is to preserve the baseline configuration of dapr sidecar and disallow the app from changing this at runtime. App will not be able to unload components that were loaded as part of initial configuration.

Reply:
Got it,
In that case, wouldn't be better to use something policy-based per component like we do for [denying](https://github.com/dapr/dapr/blob/master/pkg/config/configuration.go#L227) components?

Reply:
added a CLI flag to enable dynamic loading for the same directory `components-path`, instead of having a separate path arg.